check_api/src/main/java/com/google/errorprone/scanner/ScannerSupplier.java

```diff
@@ -298,7 +299,7 @@ public ScannerSupplier plus(ScannerSupplier other) {
   public ScannerSupplier filter(Predicate<? super BugCheckerInfo> predicate) {
     ImmutableSet<String> disabled =
         getAllChecks().values().stream()
-            .filter(predicate.negate())
+            .filter(bci -> disabled().contains(bci.canonicalName()) || !predicate.test(bci))
```

Comment:
The changes in #4699 assumed that `ScannerSupplier#filter` would merely disable additional checks, but in fact it could re-enable disabled checks. This change proposes to resolve that, yielding more intuitive semantics.